File: code/Java/interpreter/command/WhileCommand.java
package interpreter.command;

import interpreter.expr.BoolExpr;

public class WhileCommand extends Command {

    private BoolExpr cond;
    private Command cmds;

    public WhileCommand(int line, BoolExpr cond, Command cmds) {
        super(line);
        this.cond = cond;
        this.cmds = cmds;
    }

    public void execute() {
        while (cond.expr())
            cmds.execute();
    }

}

File: code/C++/interpreter/command/WhileCommand.cpp
#include "WhileCommand.h"
#include "../expr/BoolExpr.h"

WhileCommand::WhileCommand(int line, BoolExpr* cond, Command* cmds)
	: Command(line), m_cond(cond), m_cmds(cmds) {
}

WhileCommand::~WhileCommand() {
	delete m_cond;
	delete m_cmds;
}

void WhileCommand::execute() {
	while (m_cond->expr())
		m_cmds->execute();
}
